pkg/controllers/edgeconnect/controller.go

```diff
@@ -132,38 +133,81 @@ func (controller *Controller) reconcileEdgeConnectDeletion(ctx context.Context,
 
 	edgeConnect.ObjectMeta.Finalizers = nil
 	if err := controller.client.Update(ctx, edgeConnect); err != nil {
-		_log.Debug("updating the EdgeConnect object failed")
+		_log.Debug("reconcile deletion: updating the EdgeConnect object failed")
 
 		return errors.WithStack(err)
 	}
 
 	edgeConnectClient, err := controller.buildEdgeConnectClient(ctx, edgeConnect)
 	if err != nil {
-		_log.Debug("building EdgeConnect client failed")
+		_log.Debug("reconcile deletion: building EdgeConnect client failed")
```

Comment:
if you are constantly adding the same prefix (`reconcile deletion:`), it might make sense to update the `_log := log.WithValues(...)`, to have this in it instead. Like `"scenario: "deletion"` or something.

(for the first 3 logs u use the prefix, but for the rest you don't, kind inconsistent 😅 )